src/components/@shared/FormInput/InputElement/ConsumerParameters/DefaultInput.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { ReactElement } from 'react'
+import Input, { InputProps } from '../..'
+import { Field, useField } from 'formik'
+import { ConsumerParameter } from '@components/Publish/_types'
+
+export default function DefaultInput({
+  index,
+  inputName,
+  ...props
+}: InputProps & {
+  index: number
+  inputName: string
+}): ReactElement {
+  const [field] = useField<ConsumerParameter[]>(inputName)
+
+  const getStringOptions = (options: { [key: string]: string }[]): string[] => {
+    if (!options?.length) return []
+
+    return options.map((option) => Object.keys(option)[0])
+  }
+
+  return (
+    <Field
+      {...props}
+      required={field.value[index].required === 'required'}
```

Comment:
issue(blocking): for me the `default` field was not always marked and handled as required while testing. We need to check this.
It seems to be related to this line. For publish & edit we always want input for the default value. Regardless of input on the `required` field, as it is required when following the docs:
![image](https://github.com/deltaDAO/Ocean-Market/assets/28757404/2fb9d3c5-da63-44fa-b747-cdf046d7f01f)

Reply:
Yes, I added this when we talked about letting the user input empty values for the "optional" parameters but now I see it's only for consumption.